Support V6E for type of accelerator_config in google_tpu_v2_vm

zeleena opened the pull request:
```release-note:none
tpuv2: changed `accelerator_config.tpye` field from enum to string in `google_tpu_v2_vm` resource (beta)
```

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-12353-old..auto-pr-12353) ( 1 file changed, 1 insertion(+), 1 deletion(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-12353-old..auto-pr-12353) ( 2 files changed, 3 insertions(+), 3 deletions(-))

modular-magician commented:
#### Tests analytics
Total tests: 5
Passed tests: 3
Skipped tests: 2
Affected tests: 0

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>tpuv2</li>

</ul>

</blockquote>
</details>

🟢 All tests passed!

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12353/artifacts/06b2bd8d-d347-4115-b079-22044837b810/build-log/replaying_test.log)

shuyama1 reviewed (commented):
The change looks good overall! I just have a quick question: the value doesn’t appear in the API documentation—has support for this value already rolled out?

Would you mind either adding a new test or updating an existing test to include this value? If you’ve already tested it locally and confirmed that it works, that should be fine as well.

https://googlecloudplatform.github.io/magic-modules/develop/test/test/ and https://googlecloudplatform.github.io/magic-modules/develop/test/run-tests/ in case it helps.

zeleena commented:
Thank you, this value is in Public Preview API (alpha for us) but not GA, and no changes to the API are expected between Public Preview and GA. According to Terraform Launch Stages, it looks like this can be added to `google-beta` until GA API Launch and should be added around now in order to meet deadlines.

V6E is very capacity constrained so adding a new automated test would result in a lot of stockout errors. I did set up a manual test, but tpuv2 uses the GA API in `google-beta` since we do not have a beta API. I verified that the correct request gets sent to the control plane, but actual creation would only succeed once this feature reaches GA, and we want to have it ready by GA.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-12353-old..auto-pr-12353) ( 1 file changed, 1 insertion(+), 2 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-12353-old..auto-pr-12353) ( 2 files changed, 5 insertions(+), 7 deletions(-))

modular-magician commented:
#### Tests analytics
Total tests: 5
Passed tests: 3
Skipped tests: 2
Affected tests: 0

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>tpuv2</li>

</ul>

</blockquote>
</details>

🟢 All tests passed!

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-12353/artifacts/21f87685-8ee0-41a8-b4bb-996f42bef7df/build-log/replaying_test.log)

zeleena commented:
Like we discussed offline, I changed the type of accelerator_config from Enum to String to be more robust against additions of more types in the future and linked the documentation of types that can be selected. Please let me know what you think.

I tested this change manually using the Basic Example modified to use accelerator config, and changed a field which lead to more capacity and a successful test.

shuyama1 reviewed (approved):
LGTM! Thank you!
